agent-transport/src/jvmTest/kotlin/com/epam/drill/agent/transport/QueuedAgentMessageMetadataSenderTest.kt

```diff
@@ -0,0 +1,343 @@
+import io.mockk.ManyAnswersAnswer
+import io.mockk.MockKAnnotations
+import io.mockk.ThrowingAnswer
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.verify
+
+class QueuedAgentMessageMetadataSenderTest {
+
+    private class TestAgentMessage(val msg: String) : AgentMessage()
+
+    @MockK
+    private lateinit var messageTransport: AgentMessageTransport<String>
+    @MockK
+    private lateinit var responseStatus: ResponseStatus
+    @MockK
+    private lateinit var messageSerializer: AgentMessageSerializer<AgentMessage, String>
+    @MockK
+    private lateinit var destinationMapper: AgentMessageDestinationMapper
+    @MockK
+    private lateinit var configSender: AgentMetadataSender<String>
+    @MockK
+    private lateinit var stateNotifier: TransportStateNotifier
+    @MockK
+    private lateinit var stateListener: TransportStateListener
+    @MockK
+    private lateinit var messageQueue: InMemoryAgentMessageQueue<String>
+
+    private lateinit var sender: QueuedAgentMessageSender<AgentMessage, String>
+
+    private val incomingMessage = mutableListOf<TestAgentMessage>()
+    private val incomingDestinations = mutableListOf<AgentMessageDestination>()
+    private val toSendMessages = mutableListOf<String>()
+    private val toSendContentTypes = mutableListOf<String>()
+    private val toSendDestinations = mutableListOf<AgentMessageDestination>()
+    private val queuedPairs = mutableListOf<Pair<AgentMessageDestination, String>>()
+
+    @BeforeTest
+    fun setup() = MockKAnnotations.init(this).also {
+        val serialize: (TestAgentMessage) -> String = {
+            "serialized-${it.msg}"
+        }
+        val mapDestination: (AgentMessageDestination) -> AgentMessageDestination = {
+            AgentMessageDestination("MAP", "mapped-${it.target}")
+        }
+        every { messageSerializer.serialize(capture(incomingMessage)) } answers
+                FunctionAnswer { serialize(it.invocation.args[0] as TestAgentMessage) }
+        every { messageSerializer.contentType() } returns "test/test"
+        every { destinationMapper.map(capture(incomingDestinations)) } answers
+                FunctionAnswer { mapDestination(it.invocation.args[0] as AgentMessageDestination) }
+        every { configSender.addStateListener(any()) } returns Unit
+        every { stateNotifier.addStateListener(any()) } returns Unit
+        every { stateListener.onStateFailed() } returns Unit
+        every { messageQueue.offer(capture(queuedPairs)) } returns true
+
+        sender = QueuedAgentMessageMetadataSender(messageTransport, messageSerializer, destinationMapper, configSender,
+            stateNotifier, stateListener, messageQueue)
+
+        incomingMessage.clear()
+        incomingDestinations.clear()
+        toSendMessages.clear()
+        toSendContentTypes.clear()
+        toSendDestinations.clear()
+        queuedPairs.clear()
+    }
+
+    @Test
+    fun `available only after config sent`() {

```

Comment:
We don't need the following 3 test cases since it does not matter anymore if we send data before or after AgentConfig
`available only after config sent`
`sending successful after config sent`
`sending queued before config sent`

Reply:
we'll get rid of AgentMessageMetadata sender